source/shared/core_sqlsrv.h

```diff
@@ -2283,4 +2331,20 @@ sqlsrv_conn* allocate_conn( SQLHANDLE h, error_callback e, void* driver TSRMLS_D
 
 } // namespace core
 
+// connection attribute functions
+template <unsigned int Attr>
+struct str_conn_attr_func {
+
```

Comment:
because str_conn_attr_func in not defined in core namespace. it should be after core is defined otherwise GCC complains about it